### PR TITLE
dev-scripts: run agent install job in CI

### PR DESCRIPTION
**Pull request by zaneb**

### CODE REVIEW

**Comment by zaneb** [comment]
/hold
Maybe openshift-metal3/dev-scripts#1405 will help.

File: ci-operator/config/openshift-metal3/dev-scripts/openshift-metal3-dev-scripts-master.yaml [hunk 1]
```diff
@@ -62,31 +62,51 @@ tests:
...
   steps:
     cluster_profile: equinix-ocp-metal
     workflow: baremetalds-e2e-proxy-ipv4
 - as: e2e-metal-ipi-proxy-ipv6
-  skip_if_only_changed: (^(agent|docs|.github)/)|^config_example.sh$|\.md$|^(DCO|LICENSE|OWNERS)$
+  skip_if_only_changed: (^(agent|docs|.github)/)|^config_example.sh$|\.md$|^(DCO|LICENSE)$|((^|/)OWNERS$)
   steps:
     cluster_profile: equinix-ocp-metal
     workflow: baremetalds-e2e-proxy-ipv6
+- as: e2e-agent-compact
```
Comment (andfasano) [inline ci-operator/config/openshift-metal3/dev-scripts/openshift-metal3-dev-scripts-master.yaml:90]: This sounds as a good occasion to use the recently merged [agent compact workflow](https://github.com/openshift/release/blob/891e3bfe9294494fca96ead8be94c3fc85df54f6/ci-operator/step-registry/agent/e2e/compact/ipv4/agent-e2e-compact-ipv4-workflow.yaml#L2)

Comment (zaneb) [reply ci-operator/config/openshift-metal3/dev-scripts/openshift-metal3-dev-scripts-master.yaml:93]: Gave it a go but it doesn't have KNI_DEPLOY_FROM_GIT set :frowning_face:. We may need separate workflows for testing the installer and everything else until we are merged in the installer master branch.
In the meantime I copied the contents here - the predefined chains definitely make it simpler.

File: ci-operator/config/openshift-metal3/dev-scripts/openshift-metal3-dev-scripts-master.yaml [hunk 2]
```diff
@@ -62,34 +62,49 @@ tests:
...
+  skip_if_only_changed: (^docs|.github/)|^config_example.sh$|\.md$|^(DCO|LICENSE)$|((^|/)OWNERS$)
+  steps:
+    cluster_profile: equinix-ocp-metal
+    env:
+      DEVSCRIPTS_CONFIG: |
+        AGENT_E2E_TEST_SCENARIO=COMPACT_IPV4
+        KNI_DEPLOY_FROM_GIT=yesplease
```
Comment (andfasano) [inline ci-operator/config/openshift-metal3/dev-scripts/openshift-metal3-dev-scripts-master.yaml:100]:
```suggestion
        KNI_INSTALL_FROM_GIT=yesplease
```

File: ci-operator/config/openshift-metal3/dev-scripts/openshift-metal3-dev-scripts-master.yaml (hunk 1 above)
Comment (andfasano) [reply ci-operator/config/openshift-metal3/dev-scripts/openshift-metal3-dev-scripts-master.yaml:93]: Ack. If the `DEVSCRIPTS_CONFIG` it's the only part to be overriden (and unfortunately cannot be done partially), and the steps sequence remains exactly the same, then it should work also invoking directly a ` workflow: agent-e2e-compact-ipv4` (just to save a few lines of code)

File: ci-operator/config/openshift-metal3/dev-scripts/openshift-metal3-dev-scripts-master.yaml (hunk 2 above)
Comment (zaneb) [reply ci-operator/config/openshift-metal3/dev-scripts/openshift-metal3-dev-scripts-master.yaml:100]: :facepalm:

File: ci-operator/config/openshift-metal3/dev-scripts/openshift-metal3-dev-scripts-master.yaml (hunk 1 above)
Comment (zaneb) [reply ci-operator/config/openshift-metal3/dev-scripts/openshift-metal3-dev-scripts-master.yaml:93]: Done, thanks for the tip!

**Comment by zaneb** [comment]
Even though the cluster failed to come up, it does look like this job is running the way it is supposed to. I suspect that the whole world is broken right now.

**Pull request merged**